### ISSUE TITLE
MAP=0，coco_metric, cocoeval

### ISSUE THREAD

**Issue by 52THANOS** [open-mmlab/mmpose#1964]
My ap values are always 0.  My json files are correct. When i debugged the program, i found that my res_file ,coco_det and sigmas are normal in coco_metric.py. But once the programm runs in cocoeval.py，my dt values are all empty. I guess that is the main reason casuing the 0 problem. 
HOW CAN I SOLVE THIS PROBLEM?

**Comment by Tau-J** [comment]
Sorry, we cannot locate the problem unless you can provide more information. This issue looks similar to yours, maybe you can have a check: https://github.com/open-mmlab/mmpose/issues/1795#issuecomment-1340656707

**Comment by 52THANOS** [comment]
``_base_ = ['_base_/default_runtime.py',
          "_base_/datasets/T.py"]

# runtime
train_cfg = dict(max_epochs=420, val_interval=10)

# optimizer
optim_wrapper = dict(optimizer=dict(
    type='Adam',
    lr=5e-4,
))

# learning policy
param_scheduler = [
    dict(
        type='LinearLR', begin=0, end=500, start_factor=0.001,
        by_epoch=False),  # warm-up
    dict(
        type='MultiStepLR',
        begin=0,
        end=210,
        milestones=[170, 200],
        gamma=0.1,
        by_epoch=True)
]

# automatically scaling LR based on the actual training batch size
auto_scale_lr = dict(base_batch_size=512)

# hooks
default_hooks = dict(checkpoint=dict(save_best='coco/AP', rule='greater'))

# codec settings
codec = dict(
    type='MSRAHeatmap',
    input_size=(288, 384),
    heatmap_size=(72, 96),
    sigma=3,
    unbiased=True)

# model settings
model = dict(
    type='TopdownPoseEstimator',
    data_preprocessor=dict(
        type='PoseDataPreprocessor',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        bgr_to_rgb=True),
    backbone=dict(
        type='HRNet',
        in_channels=3,
        extra=dict(
            stage1=dict(
                num_modules=1,
                num_branches=1,
                block='BOTTLENECK',
                num_blocks=(4, ),
                num_channels=(64, )),
            stage2=dict(
                num_modules=1,
                num_branches=2,
                block='BASIC',
                num_blocks=(4, 4),
                num_channels=(48, 96)),
            stage3=dict(
                num_modules=4,
                num_branches=3,
                block='BASIC',
                num_blocks=(4, 4, 4),
                num_channels=(48, 96, 192)),
            stage4=dict(
                num_modules=3,
                num_branches=4,
                block='BASIC',
                num_blocks=(4, 4, 4, 4),
                num_channels=(48, 96, 192, 384))),
        init_cfg=dict(
            type='Pretrained',
            checkpoint='https://download.openmmlab.com/mmpose/'
            'pretrain_models/hrnet_w48-8ef0771d.pth'),
    ),
    head=dict(
        type='HeatmapHead',
        in_channels=48,
        out_channels=22,
        deconv_out_channels=None,
        loss=dict(type='KeypointMSELoss', use_target_weight=True),
        decoder=codec),
    test_cfg=dict(
        flip_test=True,
        flip_mode='heatmap',
        shift_heatmap=True,
    ))

# base dataset settings
dataset_type = 'TDataset'
data_mode = 'topdown'
data_root = 'test_coco_data/coco/'

# pipelines
train_pipeline = [
    dict(type='LoadImage', file_client_args={{_base_.file_client_args}}),
    dict(type='GetBBoxCenterScale'),
    dict(type='RandomFlip', direction='horizontal'),
    dict(type='RandomHalfBody'),
    dict(type='RandomBBoxTransform'),
    dict(type='TopdownAffine', input_size=codec['input_size']),
    dict(type='GenerateTarget', target_type='heatmap', encoder=codec),
    dict(type='PackPoseInputs')
]
val_pipeline = [
    dict(type='LoadImage', file_client_args={{_base_.file_client_args}}),
    dict(type='GetBBoxCenterScale'),
    dict(type='TopdownAffine', input_size=codec['input_size']),
    dict(type='PackPoseInputs')
]

# data loaders
train_dataloader = dict(
    batch_size=32,
    num_workers=2,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=True),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        data_mode=data_mode,
        ann_file='annotations/train2017.json',
        data_prefix=dict(img='train2017/'),
        pipeline=train_pipeline,
    ))
val_dataloader = dict(
    batch_size=32,
    num_workers=2,
    persistent_workers=True,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False, round_up=False),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        data_mode=data_mode,
        ann_file='annotations/val2017.json',
        # bbox_file='data/coco/person_detection_results/'
        # 'COCO_val2017_detections_AP_H_56_person.json',
        data_prefix=dict(img='val2017/'),
        test_mode=True,
        pipeline=val_pipeline,
    ))
test_dataloader = val_dataloader

# evaluators
val_evaluator = dict(
    type='CocoMetric',
    ann_file=data_root + 'annotations/val2017.json')
test_evaluator = val_evaluator

**Comment by 52THANOS** [comment]
``dataset_info = dict(
    dataset_name='coco',
    paper_info=dict(
        author='Lin, Tsung-Yi and Maire, Michael and '
        'Belongie, Serge and Hays, James and '
        'Perona, Pietro and Ramanan, Deva and '
        r'Doll{\'a}r, Piotr and Zitnick, C Lawrence',
        title='Microsoft coco: Common objects in context',
        container='European conference on computer vision',
        year='2014',
        homepage='http://cocodataset.org/',
    ),
    keypoint_info={
        0:
        dict(
            name='femeral_notch_one',
            id=7,
            color=[255, 128, 0],
            type='lower',
            swap='femeral_notch_two'),
        1:
        dict(
            name='femeral_notch_two',
            id=8,
            color=[255, 128, 0],
            type='lower',
            swap='femeral_notch_one'),
        2:
        dict(
            name='femoral_condyles_one',
            id=9,
            color=[0, 255, 0],
            type='lower',
            swap='femoral_condyles_three'),
        3:
        dict(
            name='femoral_condyles_two',
            id=10,
            color=[0, 255, 0],
            type='lower',
            swap='femoral_condyles_four'),
        4:
        dict(
            name='femoral_condyles_three',
            id=11,
            color=[0, 255, 0],
            type='lower',
            swap='femoral_condyles_one'),
        5:
        dict(
            name='femoral_condyles_four',
            id=12,
            color=[0, 255, 0],
            type='lower',
            swap='femoral_condyles_two'),
        6:
        dict(
            name='tibial_spines_one',
            id=13,
            color=[0, 255, 128],
            type='lower',
            swap='tibial_spines_two'),
        7:
        dict(
            name='tibial_spines_two',
            id=14,
            color=[0, 255, 128],
            type='lower',
            swap='tibial_spines_one'),
        8:
        dict(
            name='center_tibia_one',
            id=15,
            color=[255, 0, 0],
            type='lower',
            swap='center_tibia_three'),
        9:
        dict(
            name='center_tibia_two',
            id=16,
            color=[255, 0, 0],
            type='lower',
            swap='center_tibia_four'),
        10:
        dict(
            name='center_tibia_three',
            id=16,
            color=[255, 0, 0],
            type='lower',
            swap='center_tibia_one'),
        11:
        dict(
            name='center_tibia_four',
            id=16,
            color=[255, 0, 0],
            type='lower',
            swap='center_tibia_two'),
        12:
        dict(
            name='talus_one',
            id=16,
            color=[255, 0, 128],
            type='lower',
            swap='talus_three'),
        13:
        dict(
            name='talus_two',
            id=16,
            color=[255, 0, 128],
            type='lower',
            swap='talus_four'),
        14:
        dict(
            name='talus_three',
            id=16,
            color=[255, 0, 128],
            type='lower',
            swap='talus_one'),
        15:
        dict(
            name='talus_four',
            id=16,
            color=[255, 0, 128],
            type='lower',
            swap='talus_two'),
        16:
        dict(
            name='fhc_one',
            id=16,
            color=[255, 128, 128],
            type='lower',
            swap='fhc_two'),
        17:
        dict(
            name='fhc_two',
            id=16,
            color=[255, 128, 128],
            type='lower',
            swap='fhc_one'),
        18:
        dict(
            name='femoral_lateral_one',
            id=16,
            color=[255, 128, 255],
            type='lower',
            swap='femoral_lateral_two'),
        19:
        dict(
            name='femoral_lateral_two',
            id=16,
            color=[255, 128, 255],
            type='lower',
            swap='femoral_lateral_one'),
        20:
        dict(
            name='femoral_medial_one',
            id=16,
            color=[255, 128, 255],
            type='lower',
            swap='femoral_medial_two'),
        21:
        dict(
            name='femoral_medial_two',
            id=16,
            color=[255, 128, 255],
            type='lower',
            swap='femoral_medial_one'),

    },
    skeleton_info={
        # 0:
        # dict(link=('left_ankle', 'left_knee'), id=0, color=[0, 255, 0]),
        # 1:
        # dict(link=('left_knee', 'left_hip'), id=1, color=[0, 255, 0]),
        # 2:
        # dict(link=('right_ankle', 'right_knee'), id=2, color=[255, 128, 0]),
        # 3:
        # dict(link=('right_knee', 'right_hip'), id=3, color=[255, 128, 0]),
        # 4:
        # dict(link=('left_hip', 'right_hip'), id=4, color=[51, 153, 255]),
        # 5:
        # dict(link=('left_shoulder', 'left_hip'), id=5, color=[51, 153, 255]),
        # 6:
        # dict(link=('right_shoulder', 'right_hip'), id=6, color=[51, 153, 255]),
        # 7:
        # dict(
        #     link=('left_shoulder', 'right_shoulder'),
        #     id=7,
        #     color=[51, 153, 255]),
        # 8:
        # dict(link=('left_shoulder', 'left_elbow'), id=8, color=[0, 255, 0]),
        # 9:
        # dict(
        #     link=('right_shoulder', 'right_elbow'), id=9, color=[255, 128, 0]),
        # 10:
        # dict(link=('left_elbow', 'left_wrist'), id=10, color=[0, 255, 0]),
        # 11:
        # dict(link=('right_elbow', 'right_wrist'), id=11, color=[255, 128, 0]),
        # 12:
        # dict(link=('left_eye', 'right_eye'), id=12, color=[51, 153, 255]),
        # 13:
        # dict(link=('nose', 'left_eye'), id=13, color=[51, 153, 255]),
        # 14:
        # dict(link=('nose', 'right_eye'), id=14, color=[51, 153, 255]),
        # 15:
        # dict(link=('left_eye', 'left_ear'), id=15, color=[51, 153, 255]),
        # 16:
        # dict(link=('right_eye', 'right_ear'), id=16, color=[51, 153, 255]),
        # 17:
        # dict(link=('left_ear', 'left_shoulder'), id=17, color=[51, 153, 255]),
        # 18:
        # dict(
        #     link=('right_ear', 'right_shoulder'), id=18, color=[51, 153, 255])
    },
    joint_weights=[
        1., 1., 1., 1., 1.,
        1., 1., 1.2, 1.2, 1.5,
        1.5, 1., 1., 1.2, 1.2,
        1.5, 1.5, 1.,
        # 1., 1., 1., 1., 1., 1.,
    ],
    sigmas=[
        0.026, 0.025, 0.025, 0.035, 0.035,
        0.079, 0.079, 0.072, 0.072, 0.062,
        0.062, 0.107, 0.107, 0.087, 0.087,
        0.089, 0.089, 0.062,
        # 0.107, 0.107, 0.087, 0.087, 0.089, 0.089,

    ])

**Comment by 52THANOS** [comment]
> Sorry, we cannot locate the problem unless you can provide more information. This issue looks similar to yours, maybe you can have a check: [#1795 (comment)](https://github.com/open-mmlab/mmpose/issues/1795#issuecomment-1340656707)

this is my content of coco_det in cocometric.
1: {'category_id': 1, 'image_id': 1665, 'keypoints': [144.96191009879112, 536.5314024984837, 0.8084061145782471, 390.71750938892365, 523.5054193735123, 0.7580474615097046, 129.55135390162468, 536.8023473322392, 0.8437880873680115, 162.21052192151546, 538.3749649226665, 0.777603268623352, 407.3760664463043, 523.45210942626, 0.7913923263549805, 373.7784416973591, 526.0320316553116, 0.7356181740760803, 148.15189771354198, 538.880401134491, 0.8250700235366821, 387.23317950963974, 526.1844639778137, 0.784142255783081, 168.22327382862568, 538.5503034591675, 0.6920623779296875, 125.11734686791897, 538.5559395253658, 0.7670177221298218, 368.9584158360958, 526.5381400585175, 0.662995457649231, 411.8291275203228, 525.4393656849861, 0.7181109189987183, 235.61146810650826, 931.5201069712639, 0.31969153881073, 240.34571200609207, 931.7168005108833, 0.4261004328727722, 245.57479217648506, 931.8272467255592, 0.30786585807800293, 241.26170203089714, 931.6657139658928, 0.417944997549057, 122.18873828649521, 93.38419283926487, 0.30419421195983887, 123.25470454990864, 92.91508081555367, 0.26808857917785645, 122.31146492063999, 531.4380015730858, 0.7400121092796326, 415.99542140960693, 517.2957153916359, 0.679262101650238, 171.13626688718796, 528.7398253679276, 0.6055866479873657, 366.8179516494274, 516.1365989148617, 0.5537739992141724], 'score': 0.6266715756871484, 'area': 246478.00089764435, 'id': 1, 'bbox': [122.18873828649521, 92.91508081555367, 293.8066831231117, 838.9121659100056]},
My detection results have turned into json file format like gt json file. Is there something wrong with my content?
